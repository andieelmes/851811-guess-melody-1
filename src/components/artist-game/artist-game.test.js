import React from 'react';
import renderer from 'react-test-renderer';
import ArtistGame from 'components/artist-game/artist-game.jsx';

const gameInfo = {
  id: `a`,
  type: `artist`,
  song: {
    artist: `Jim Beam`,
    src: `path.mp3`,
  },
  answers: [
    {
      id: `b`,
      picture: `https://static.tvtropes.org/pmwiki/pub/images/jon_snow_s7.png`,
      artist: `John Snow`,
    },
  ],
};

describe(`Artist game component`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
      .create(<ArtistGame {...gameInfo} onClick={() => {}}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
