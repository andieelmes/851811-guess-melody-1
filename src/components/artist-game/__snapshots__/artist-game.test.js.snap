// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist game component should render correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <audio
        src="path.mp3"
      />
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-b"
          name="answer"
          onClick={[Function]}
          type="radio"
          value="artist-b"
        />
        <label
          className="artist__name"
          htmlFor="answer-b"
        >
          <img
            alt="John Snow"
            className="artist__picture"
            src="https://static.tvtropes.org/pmwiki/pub/images/jon_snow_s7.png"
          />
          John Snow
        </label>
      </div>
    </form>
  </section>
</section>
`;
